src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,57 @@
+import globalsJson from '../globals.json';
+import { createRule, parseJestFnCall } from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Prefer importing Jest globals',
+      recommended: false,
+    },
+    messages: {
+      preferImportingJestGlobal:
+        "Jest function \"{{ jestFunction }} is used but not imported from '@jest/globals'",
+    },
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const jestGlobalFunctions = Object.keys(globalsJson);
+    const importedJestFunctions: string[] = [];
+    const usedJestFunctions = new Set<string>();
+
+    return {
+      CallExpression(node) {
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) {
+          return;
+        }
+        if (
+          jestFnCall.head.type === 'import' &&
+          jestGlobalFunctions.includes(jestFnCall.name)
+        ) {
+          importedJestFunctions.push(jestFnCall.name);
+        }
+
+        /* istanbul ignore else */
+        if (jestGlobalFunctions.includes(jestFnCall.name)) {
+          usedJestFunctions.add(jestFnCall.name);
+        }
+      },
+      'Program:exit'() {
+        usedJestFunctions.forEach(jestFunction => {
+          if (!importedJestFunctions.includes(jestFunction)) {
+            context.report({
```

Comment:
We should report when we find a non-imported jest function call both for efficiency and it means we can flag the specific function rather than just the whole file.

Up to you if you want to keep the tracking of already reported functions - I don't think it's that big of a deal if we report on _every_ function call since it'll all get fixed by the same line, but also its not much extra code having the `Set` 🤷

Reply:
I need to report at the end `Program:exit()` to know all the Jest functions that needs to be imported & build the complete fix. If I report during the `CallExpression` I might incomplete `import` statement.

See test failing:
```
    - import { describe, test, expect } from '@jest/globals';
    + import { describe } from '@jest/globals';
```